### PR TITLE
Copy AccessLists between doors and boards.

### PR DESCRIPTION
**Pull request by Vordenburg**
## About the PR 

The door electronics board for an airlock does not factor into its access. This PR fixes that.

This solves half of the issue mentioned in #5353. Another PR will have to deal with programming new door electronics boards.

**Screenshots**

**Changelog**

:cl:
- fix: Deconstructing an airlock will now give you a door electronics board with its access programmed accordingly to the airlock.

### CODE REVIEW

File: Content.Server/Doors/Systems/DoorSystem.cs [hunk 1]
```diff
@@ -280,6 +282,56 @@ private void OnEmagged(EntityUid uid, DoorComponent door, GotEmaggedEvent args)
...
+    }
+
+    private void OnRemoved(EntityUid uid, DoorComponent component, ContainerModifiedMessage args)
+    {
+        if (args.Container.ID == "board" &&
+            TryComp<AccessReaderComponent>(args.Entity, out var ejectedReaderComponent) &&
+            TryComp<AccessReaderComponent>(uid, out var readerComponent))
+        {
+            // Copy the airlock's access lists to the board.
+            ejectedReaderComponent.AccessLists = readerComponent.AccessLists;
```
Comment (metalgearsloth) [inline Content.Server/Doors/Systems/DoorSystem.cs:327]: As above.

File: Content.Server/Doors/Systems/DoorSystem.cs [hunk 2]
```diff
@@ -280,6 +282,56 @@ private void OnEmagged(EntityUid uid, DoorComponent door, GotEmaggedEvent args)
...
+        if (args.Container.ID != "board" ||
+            !TryComp<AccessReaderComponent>(args.Entity, out var insertedReaderComponent))
+        {
+            return;
+        }
+
+        // We check for 1 and not 0 because the board's already in the container at this point.
+        if (_containerSystem.GetContainer(uid, "board").ContainedEntities.Count != 1)
+        {
+            Logger.Warning($"Attempted to insert a second board {ToPrettyString(args.Entity)} into door {ToPrettyString(uid)}!");
```
Comment (metalgearsloth) [inline Content.Server/Doors/Systems/DoorSystem.cs:296]: Static loggers are deprecated and you should use a sawmill instead.

File: Content.Server/Doors/Systems/DoorSystem.cs [hunk 3]
```diff
@@ -280,6 +282,56 @@ private void OnEmagged(EntityUid uid, DoorComponent door, GotEmaggedEvent args)
...
+            insertedReaderComponent.AccessLists = readerComponent.AccessLists;
+        }
+        else
+        {
+            // Otherwise, copy the board's access onto the door.
+            // This path will be followed if the door had no access to begin
+            // with on map setup, which isn't a problem,
+            // since they'll both be blank.
+            readerComponent.AccessLists = insertedReaderComponent.AccessLists;
+        }
```
Comment (metalgearsloth) [inline Content.Server/Doors/Systems/DoorSystem.cs:317]: Having maploader specific codepaths shouldn't be done as it leads to annoying bugs and we've removed most of them. You should just be doing it on accessreader startup, at best just checking lifestage.

Also these need to be deep copied.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Doors/Systems/DoorSystem.cs (hunk 3 above)
Comment (Vordenburg) [reply Content.Server/Doors/Systems/DoorSystem.cs:317]: At present, I'm unable to figure out how to both support inserting a door electronics board that has new access into an airlock and the copying of map access onto a board at map load without somehow making the code aware of map loading, given the door electronics are going through component startup before the doors, and the insertion event is firing on map load. It's a chicken and egg problem.

**Comment by AJCM-git** [comment]
Reopen once you are free

**Pull request closed without merging**